contrib/systemd/etcd2-backup-coreos/README.md

```diff
@@ -0,0 +1,255 @@
+
+If you want to adjust backup frequency, edit `./etcd2-backup.timer`
+
+## Installation
+Once you've got those things configured, you can run `./build`.
+
+That script will generate a tarball can be copied to CoreOS instances and contains a script `etcd2-backup-install`
+
+After the tarball is inflated and the install script exected, you should see three new systemd services - one used for periodically backing up and two for restoring the cluster.
+
+- `etcd2-backup.service`
+   A oneshot service which calls `etcdctl backup` and syncs the backups to the rclone endpoint (using an rclone container, of course). `etcd2-backup.timer` is responsible for periodically running this service.
+
+- `etcd2-restore.service`
+    A oneshot service which wipes all etcd2 data and restores a single-node cluster from the rclone backup. This is for restoring the **founding member** only.
+
+- `etcd2-join.service`
+   A oneshot service which wipes all etcd2 data and re-joins the new cluster. This is for adding members **after** the **founding member** has succesfully established the new cluster via `etcd2-restore.service`
+
+## Recovery
+
+This assumes that your cluster has lost majority, and is not recoverable. Otherwise you should probably try to heal your cluster first.
+
+There are a two factors which contribute to how "close" to up-to-date your backup is.
+
+- `etcd2-backup.timer` is configured by default to run a backup every 30 seconds.
+
+- etcd2 has a parameter `snapshot-count` parameter, which controls how many transactions can be committed before the snapshot is written to disk.
+
+Let's compute the "maximum outdatedness" of a backup
+
+Assumptions
+- transaction rate is a constant `1000 transactions / second`
+- `etcd2-backup.timer` is configured for a 30 second interval
+- `etcd2 snapshot-count=10000`
+
+```
+max-missed-seconds= (10000 transactions / (1000 transactions / second)) + 30 seconds = 40 seconds
+```
+
+
+1. Make sure `etcd2.service` and `etcd2-backup.timer` are stopped on all nodes in the cluster
+
+2. Restore the **founding member** by starting `etcd2-restore.service` and then, if successful, `etcd2.service`
+
+3. Restore the rest of the cluster **one at a time**. Start `etcd2-join.service`, and then, if successful, `etcd2.service`. Please verify with `etcdctl cluster-health` that the expected set of nodes is present and healthy after you join each node.
+
+4. Verify that your data is sane (enough). If so, kick off `etcd2-backup.timer` on all nodes and, hopefully, go back to bed.
+
+## Retroactively change the founding member
+
+This will need to happen if you want to restore a cluster from some node's data other than the original founding member.
+
+Change the value of `ETCD_RESTORE_MASTER_ADV_PEER_URLS` in `30-etcd2-backup-restore.conf` to the advertised peer url of the new master node. Repeat the install process, then proceed with recovery instructions.
```

Comment:
you actually need to re-do the install on all nodes. they all need to agree who the master is